htlcswitch/hop/iterator.go

```diff
@@ -113,6 +120,98 @@ func (r *sphinxHopIterator) ExtractErrorEncrypter(
 	return extracter(r.ogPacket.EphemeralKey)
 }
 
+// BlindingProcessor is an interface that provides the cryptographic operations
+// required for processing blinded hops.
+type BlindingProcessor interface {
+	// DecryptBlindedHopData decrypts a blinded blob of data using the
+	// ephemeral key provided.
+	DecryptBlindedHopData(ephemPub *btcec.PublicKey,
+		encryptedData []byte) ([]byte, error)
+}
+
+// BlindingKit contains the components required to extract forwarding
+// information for hops in a blinded route.
+type BlindingKit struct {
+	// BlindingPoint holds a blinding point that was passed to the node via
+	// update_add_htlc's TLVs.
+	BlindingPoint *btcec.PublicKey
+
+	// ForwardingInfo uses the ephemeral blinding key provided to decrypt
+	// a blob of encrypted data provided in the onion and obtain the
+	// forwarding information for the blinded hop.
+	ForwardingInfo func(*btcec.PublicKey, []byte) (*ForwardingInfo,
+		error)
+
+	// DisableBlindedFwd provides the ability to optionally disable blinded
+	// route forwarding. Payloads will simply be rejected as invalid if
+	// they contain blinding points, as our node does not want to forward
+	// them.
+	DisableBlindedFwd bool
```

Comment:
Same comment about if we can lift this up to a higher layer. So we never even try to decrypt/decode something when we don't support it. 

I think not yet in the scope of these PRs, but we should also have path finding logic to not attempt to include a node in our blinded path if they don't support it.

Reply:
Eg: we don't even need to _call_ `MakeBlindingKit` (just for it to reject the payload) if we don't support the feature.

Reply:
> but we should also have path finding logic to not attempt to include a node in our blinded path if they don't support it.

Def, I imagine that would be for blinded path creation? Filter by feature bit and then make a path seems to be the norm.